tests/test_secondary.py

```diff
@@ -0,0 +1,721 @@
+"""
+<Program Name>
+  test_secondary.py
+
+<Purpose>
+  Unit testing for uptane/clients/secondary.py
+  Much of this is copied from test_primary.py and then modified slightly
+"""
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import uptane # Import before TUF modules; may change tuf.conf values.
+
+import unittest
+import os.path
+import time
+import shutil
+import hashlib
+
+from six.moves.urllib.error import URLError
+
+import tuf
+import tuf.formats
+import tuf.conf
+import tuf.client.updater
+
+import uptane.formats
+import uptane.clients.secondary as secondary
+import uptane.common # verify sigs, create client dir structure, convert key
+import uptane.encoding.asn1_codec as asn1_codec
+
+# For temporary convenience:
+import demo # for generate_key, import_public_key, import_private_key
+
+TEST_DATA_DIR = os.path.join(uptane.WORKING_DIR, 'tests', 'test_data')
+TEST_DIRECTOR_METADATA_DIR = os.path.join(TEST_DATA_DIR, 'director_metadata')
+TEST_IMAGE_REPO_METADATA_DIR = os.path.join(
+    TEST_DATA_DIR, 'image_repo_metadata')
+TEST_DIRECTOR_ROOT_FNAME = os.path.join(
+    TEST_DIRECTOR_METADATA_DIR, 'root.' + tuf.conf.METADATA_FORMAT)
+TEST_IMAGE_REPO_ROOT_FNAME = os.path.join(
+    TEST_IMAGE_REPO_METADATA_DIR, 'root.' + tuf.conf.METADATA_FORMAT)
+TEST_PINNING_FNAME = os.path.join(TEST_DATA_DIR, 'pinned.json')
+TEMP_CLIENT_DIR_1 = os.path.join(TEST_DATA_DIR, 'temp_test_secondary1')
+TEMP_CLIENT_DIR_2 = os.path.join(TEST_DATA_DIR, 'temp_test_secondary2')
+TEMP_CLIENT_DIR_3 = os.path.join(TEST_DATA_DIR, 'temp_test_secondary3')
+
+# I'll initialize these in the __init__ test, and use this for the simple
+# non-damaging tests so as to avoid creating objects all over again.
+secondary_instance_1 = None # for ECU 'TCUdemocar' in vehicle 'democar'
+secondary_instance_2 = None # for an unknown ECU in vehicle '000'
+secondary_instance_3 = None # for an unknown ECU in vehicle '000'
+
+# Changing these values would require producing new signed test data from the
+# Timeserver (in the case of nonce) or a Secondary (in the case of the others).
+nonce = 5
+vin1 = 'democar'
+vin2 = 'democar'
+vin3 = '000'
+ecu_serial1 = 'TCUdemocar'
+ecu_serial2 = '00000'
+ecu_serial3 = '00000'
+
+# Initialize these in setUpModule below.
+secondary_ecu_key = None
+key_timeserver_pub = None
+key_timeserver_pri = None # simplifies use of test data (re-sign)
+key_directortargets_pub = None
+clock = None
+
+# Set starting firmware fileinfo (that this ECU had coming from the factory)
```

Comment:
Adding note `# It will serve as the initial firmware state for the Secondary clients.`.